docs/docker/configuration.md

```diff
@@ -0,0 +1,72 @@
+# Configuration
+
+The recommended way to configure the Pi-hole docker contianer is by utilising [environment variables](https://docs.docker.com/compose/how-tos/environment-variables/), however if you are persisting your `/etc/pihole` directory, you choose instead to set them via the web interface, or by directly editing `pihole.toml`
+
+## Environment Variables
+
+### Recommended Variables
+
+#### `TZ` (Default: `UTC`)
+
+Set your [timezone](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones) to make sure logs rotate at local midnight instead of at UTC midnight.
+
+#### `FTLCONF_webserver_api_password` (Default: `unset`)
+
+To set a specific password for the web interface, use the environment variable `FTLCONF_webserver_api_password` (per the quickstart example). If this variable is not detected, and you have not already set one previously inside the container via `pihole setpassword` or `pihole-FTL --config webserver.api.password`, then a random password will be assigned on startup, and will be printed to the log. You can find this password with the command `docker logs pihole | grep random password` on your host to find this password.
+
+#### `FTLCONF_dns_upstreams` (Default: `8.8.8.8;8.8.4.4`)
+
+- Upstream DNS server(s) for Pi-hole to forward queries to, separated by a semicolon
+- Supports non-standard ports with #[port number] e.g `127.0.0.1#5053;8.8.8.8;8.8.4.4`
+- Supports Docker service names and links instead of IPs e.g `upstream0;upstream1` where upstream0 and upstream1 are the service names of or links to docker services
+
+### Optional Variables
+
+#### `FTLCONF_[SETTING]`
+
+Set various `pihole.toml` settings via environment variable for persistence. Follows the syntax as setting a value on the CLI, e.g `pihole-FTL --config key.setting=value`. Replace `.` with `_` in the setting name, and prepend with `FTLCONF_`
+
+For example, setting `FTLCONF_dns_dnssec: 'true'` is the functional equivalent of running `pihole-FTL --config dns.dnssec=true`, or setting the following in the toml file
+
+```toml
+[dns]
+  dnssec = true
+```
+
+!!! info
```

Comment:
Maybe this should be a warning to get the users attention.

```suggestion
!!! warning
```

Also, all env variables are affected (not only the Optional ones), so I suggest to move this box to the top of the `## Environment Variables` section.

Reply:
Its supposed to be denoting that the FTLCONF_ ones become read-only. I'll make the text clearer.

Also I think info serves better than warning here, as warning denotes a mild amount of risk/danger, whereas it's supposed to be more about awareness